maya/core.py

```diff
@@ -526,6 +526,27 @@ def contains_dt(self, dt):
         return self.start <= dt < self.end
 
     @validate_arguments_type_of_function()
+    def same_length_as(self, dt):
```

Comment:
What about calling it `same_duration_as` ? The `length` is always a `duration` for this kind of interval.